ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/navigation-right/navigation-right.tsx

```diff
@@ -150,52 +151,66 @@ const toggleUser = () => {
   SlideoutRightViewInstance.open()
 }
 
+const Help = () => (
+  <Button
+    className="navigation-item"
+    icon="fa fa-question"
+    buttonType={buttonTypeEnum.neutral}
+    title="Shows helpful hints in the current context"
+    onClick={toggleHelp}
+    fadeUntilHover={true}
+  />
+)
+
+const Settings = () => (
+  <Button
+    className="navigation-item"
+    icon="fa fa-cog"
+    buttonType={buttonTypeEnum.neutral}
+    title="Shows settings for the application"
+    onClick={toggleUserSettings}
+    fadeUntilHover={true}
+  />
+)
+
+const Notifications = () => (
+  <Button
+    className="navigation-item item-alerts"
+    buttonType={buttonTypeEnum.neutral}
+    title="Shows notifications"
+    onClick={toggleAlerts}
+    fadeUntilHover={true}
+  >
+    <span className="fa fa-bell" />
+    <span className="alerts-badge fa fa-exclamation" />
+  </Button>
+)
+
+const User = () => (
+  <Button
+    className="navigation-item item-user"
+    buttonType={buttonTypeEnum.neutral}
+    onClick={toggleUser}
+    fadeUntilHover={true}
+  >
+    <div className="user-unique" title={`Logged in as ${user.getUserName()}`}>
+      <span className="">{user.getUserName()}</span>
+      <span className="fa fa-user" />
+    </div>
+    <div className="user-guest" title="Logged in as guest.">
+      <span className="">Sign In</span>
+    </div>
+  </Button>
+)
+
+const buttons = plugin([Help, Settings, Notifications, User])
+
 export default function NavigationRight(props: Props) {
   return (
     <Root {...props}>
-      <Button
-        className="navigation-item"
-        icon="fa fa-question"
-        buttonType={buttonTypeEnum.neutral}
-        title="Shows helpful hints in the current context"
-        onClick={toggleHelp}
-        fadeUntilHover={true}
-      />
-      <Button
-        className="navigation-item"
-        icon="fa fa-cog"
-        buttonType={buttonTypeEnum.neutral}
-        title="Shows settings for the application"
-        onClick={toggleUserSettings}
-        fadeUntilHover={true}
-      />
-      <Button
-        className="navigation-item item-alerts"
-        buttonType={buttonTypeEnum.neutral}
-        title="Shows notifications"
-        onClick={toggleAlerts}
-        fadeUntilHover={true}
-      >
-        <span className="fa fa-bell" />
-        <span className="alerts-badge fa fa-exclamation" />
-      </Button>
-      <Button
-        className="navigation-item item-user"
-        buttonType={buttonTypeEnum.neutral}
-        onClick={toggleUser}
-        fadeUntilHover={true}
-      >
-        <div
-          className="user-unique"
-          title={`Logged in as ${user.getUserName()}`}
-        >
-          <span className="">{user.getUserName()}</span>
-          <span className="fa fa-user" />
-        </div>
-        <div className="user-guest" title="Logged in as guest.">
-          <span className="">Sign In</span>
-        </div>
-      </Button>
+      {buttons.map((Component: any, i: number) => (
```

Comment:
Do we need this level of extensibility?  This would allow removal of the base buttons, rearranging, etc.  

Maybe it would be better to simply add a hole for downstream projects to add on to the base buttons.

Reply:
@djblue?

Reply:
This level of flexibility is nice for later if we do need to re-arrange / remove from downstream.

Reply:
This flexibility doesn't cost much in terms of complexity either which is a plus.